src/models/messages.js

```diff
@@ -14,8 +14,21 @@ function * sendMessage (message, requestingUser) {
     throw new InvalidBodyError(error, validationResult.errors)
   }
 
-  const recipientName = uri.parse(message.account, 'account').name.toLowerCase()
-  const senderAccount = uri.make('account', requestingUser.name)
+  const mode1 = !!message.account
```

Comment:
I'd refactor this section to make it so that if `account` is set, the server:

* Asserts that `from`/`to` are not set
* Rewrites the object to use `from`/`to`
* Removes the `account` property

And then proceeds with the code as normal. Reason being that I really don't want to think of the two "modes" as equals. One is the deprecated way which is included for backwards compatibility and one is the normal way. So the compatibility code should be separate (and ideally have a comment stating "for backwards compatibility".)